Lib/test/_test_multiprocessing.py

```diff
@@ -5760,6 +5760,41 @@ def test_namespace(self):
         self.run_worker(self._test_namespace, o)
 
 
+class TestNamedResource(unittest.TestCase):
+
+    def test_global_named_resource_spawn(self):
+        #
+        # Check that global named resources in main module
+        # will not leak by a subprocess, in spawn context.
+        #
+        import tempfile as tf
+        py = tf.NamedTemporaryFile('w', delete=False)
```

Comment:
Instead of creating a temporary file, you should be able to pass the source script with the `-c` option on the command line?

Reply:
As far as I know, `multiprocessing` does not support interactive code (as well as from python -c).
Mentioned in https://docs.python.org/3/library/multiprocessing.html#multiprocessing.get_context

I tried and got an error:
AttributeError: Can't get attribute 'submain' on <module '__main__' (built-in)>